src/Draco.Compiler/Internal/OptimizingIr/Model/Procedure.cs

```diff
@@ -34,9 +36,10 @@ internal sealed class Procedure : IProcedure
     private readonly Dictionary<LocalSymbol, Local> locals = new();
     private readonly List<Register> registers = new();
 
-    public Procedure(Assembly assembly, FunctionSymbol symbol)
+    public Procedure(Module declaringModule, FunctionSymbol symbol)
     {
-        this.Assembly = assembly;
+        this.DeclaringModule = declaringModule;
+        this.Assembly = declaringModule.Assembly;
```

Comment:
Since we store declaring module, we can just make assembly a derived prop, no need to store it